docs/sources/style-guide/write-for-developers/index.md

```diff
@@ -0,0 +1,148 @@
+- What are the caveats, edge cases, and side effects?
+- What is the bigger picture that is not self-evident in the code? 
+- In short, what is everything that a developer needs to know to use the code?
+
+> **Note:** It might be difficult to integrate auto-generated content with other documentation, such as relevant sections of _Get started_ guides, tutorials, or detailed code examples. You might need to ask a Docs team member to ensure that your content is properly cross-referenced.
+
+### Elements of API references
+
+Properly document the most common elements of an API reference, such as the title, parameters, return values, and so on. The following suggestions will help you to write more complete and consistent documentation:
+
+| Element                                               | Description                                                           |
+| ------------------------------------------------- | ----------------------------------------------------------------------- |
+| Title and description       | The name of the element and a description of it in one or two sentences. Put API names, classes, methods, etc. in `code font`. |
+| Syntax | The code signature that defines the element. If multiple programming languages can be used, give the syntax in each. Put the signature in `code font`. |
+| Parameters | If the element has parameters, specify their descriptions, data types, and state whether they are optional or required. Put the parameters in _italic_. |
+| Return values| If the element returns a value, describe the range of possibilities and the data type. |
+| Error codes| Describe errors or exceptions and the conditions under which they occur. |
+| Comments | Describe any important information that hasn't been previously included in the title, description, syntax, parameters, or return values. For example, you may explain non-obvious context, compare it to similar elements, or provide cautionary notes about potential gotchas. |
+
+Additional tips:
+- Remember to write concisely. Don't say "This method does this..." or "This parameter describes that..."
+- When the names of code elements are singular, don't make them plural. Instead, add a plural noun to describe them. For example, don't change `MyEvent` to `MyEvents`; refer to the `MyEvent` objects. 
+- If the element does some sort of action, start the first sentence of the description with an action verb. 
+
+## Code examples
+
+It is common for readers of documentation to skim through it to find code samples they can copy and paste and run "as is". For this reason, it is worth taking the time to ensure that whenever possible the examples you provide are production-ready.
+
+Of course, this doesn't mean you should avoid writing code examples intended merely to illustrate a point. When your intent is to explain something, it isn't necessary for your code to be runnable so long as the developer can learn from it to do something similar on their own.
```

Comment:
I am completely confused by line 77.

Reply:
Not sure why it's confusing. An example is a response from a REST API. It's provided only so the developer can compare it to the output that they get. They don't run it.

Reply:
I had to stop and read this a couple of times and still wasn't quite sure what you were trying to say.  Does the following capture what you were getting at?

```suggestion
However, it isn't necessary for every code example to be runnable in production.  Some code examples are written to illustrate a point so that the developer can learn how to do something similar on their own.
```